### ISSUE TITLE
--skip=project triggers suppressions/unused for project-rule suppressions

### ISSUE THREAD

**Issue by jakeleventhal** [biomejs/biome#9258]
## Summary

When running `biome check --skip=project`, suppressions that target project-domain diagnostics can become `suppressions/unused` warnings and fail CI with `--error-on-warnings`.

This seems surprising because skipping a domain is intended to exclude that domain's checks, but it introduces a new warning from another rule.

## Biome version

`2.4.3`

## Platform

- macOS 26.4 (arm64)

### 1) Create files

`package.json`

```json
{
  "name": "biome-repro",
  "private": true
}
```

`biome.jsonc`

```jsonc
{
  "linter": {
    "domains": {
      "project": "recommended"
    },
    "rules": {
      "correctness": {
        "noUndeclaredDependencies": "on"
      }
    }
  }
}
```

`index.ts`

```ts
// biome-ignore lint/correctness/noUndeclaredDependencies: repro for --skip=project interaction
import { foo } from "not-installed-package";

void foo;
```

### 2) Run without skipping project domain

```bash
bunx biome check index.ts --error-on-warnings
```

Result: exits `0`

### 3) Run with `--skip=project`

```bash
bunx biome check index.ts --error-on-warnings --skip=project
```

Result: exits `1` with:

- `suppressions/unused` at line 1

## Expected behavior

If `project` diagnostics are skipped, suppressions for project-domain diagnostics should not produce a new `suppressions/unused` warning (or there should be a way to keep this from failing in this mode).

## Actual behavior

`--skip=project` causes:

- `suppressions/unused` warning
- non-zero exit when `--error-on-warnings` is enabled

This makes `--skip=project` difficult to use in existing codebases that have valid suppressions for project rules.

**Comment by dyc3** [comment]
I'm thinking that it would make sense to simply not bother with suppression checks if `--skip` or `--only` is used.

**Comment by ematipico** [comment]
Yeah, but the analyser doesn't know anything about only and skip. It uses the AnalyzerFilter, which contains enabled and disabled rules. 

I have to think how we can solve it. It's weird though, can you replicate it when you run a single rule?

**Comment by dyc3** [comment]
I tried it and it appears `--only` doesn't have this problem

**Comment by ematipico** [comment]
Oh great, probably we don't correctly fill the filter, that's a good news
